### ISSUE TITLE
Some key combinations with modifiers are incorrect

### ISSUE THREAD

**Issue by CptPotato** [wez/wezterm#2098]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20220608-084332-f601c775

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Certain keys don't seem to work correctly with the <kbd>Ctrl</kbd> or <kbd>Shift</kbd> modifiers. These keys include <kbd>Space</kbd>, <kbd>Backspace</kbd> and <kbd>Enter</kbd> (maybe more).

### To Reproduce

I used [crossterm's](https://github.com/crossterm-rs/crossterm) `event-read` example, trying different inputs:

| keys pressed | resulting events |
|---|---|
| <kbd>Ctrl</kbd> <kbd>Space</kbd> | `Event: Key(KeyEvent { code: Char('2'), modifiers: SHIFT \| CONTROL })` |
| <kbd>Shift</kbd> <kbd>Enter</kbd> | `Event: Key(KeyEvent { code: Enter, modifiers: NONE })` |
| <kbd>Ctrl</kbd> <kbd>Enter</kbd> | `Event: Key(KeyEvent { code: Enter, modifiers: NONE })` |
| <kbd>Shift</kbd> <kbd>Backspace</kbd> | `Event: Key(KeyEvent { code: Backspace, modifiers: NONE })` |
| <kbd>Ctrl</kbd> <kbd>Backspace</kbd> | `Event: Key(KeyEvent { code: Backspace, modifiers: NONE })` |
| <kbd>Alt</kbd> <kbd>Backspace</kbd> | `Event: Key(KeyEvent { code: Backspace, modifiers: ALT })` ✔ |
| <kbd>Shift</kbd> <kbd>´</kbd> | nothing (this should result in a backtick \` on my keyboard layout) \* |

\* *I double checked, and the backtick character (<kbd>Shift</kbd> <kbd>´</kbd>) works with the current stable, but not on nightly. I tried with both `use_dead_keys` enabled and disabled; this might be an unrelated issue*

I also checked other terminal emulators (default cmd and powershell) and both receive the expected key events.

### Configuration

I used an empty config and was able to reproduce the issue.

### Expected Behavior

I expected the modifiers to show up in crossterm's input event test, as they do with other terminal emulators.

### Logs

<details>
<summary>pressing <kbd>Ctrl</kbd> <kbd>Space</kbd></summary>

```
09:43:55.919  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(LeftControl), modifiers: CTRL | LEFT_CTRL, phys_code: Some(LeftControl), raw_code: 17, scan_code: 29, repeat_count: 1, key_is_down: true, handled: Handled(false) }
09:43:55.921  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: LeftControl, modifiers: CTRL | LEFT_CTRL, repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(LeftControl), modifiers: CTRL | LEFT_CTRL, phys_code: Some(LeftControl), raw_code: 17, scan_code: 29, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
09:43:55.922  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=LeftControl mods=CTRL
09:43:56.023  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(Space), modifiers: CTRL | LEFT_CTRL, phys_code: Some(Space), raw_code: 32, scan_code: 57, repeat_count: 1, key_is_down: true, handled: Handled(false) }
09:43:56.024  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: Char(' '), modifiers: CTRL | LEFT_CTRL, repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(Space), modifiers: CTRL | LEFT_CTRL, phys_code: Some(Space), raw_code: 32, scan_code: 57, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
09:43:56.025  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=Char(' ') mods=CTRL
09:43:56.026  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "\0", Char(' ') CTRL
09:43:56.114  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Physical(Space), modifiers: CTRL | LEFT_CTRL, phys_code: Some(Space), raw_code: 32, scan_code: 57, repeat_count: 1, key_is_down: false, handled: Handled(false) }
09:43:56.116  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char(' '), modifiers: CTRL | LEFT_CTRL, repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(Space), modifiers: CTRL | LEFT_CTRL, phys_code: Some(Space), raw_code: 32, scan_code: 57, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
09:43:56.116  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char(' ') mods=CTRL
09:43:56.175  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, phys_code: Some(LeftControl), raw_code: 17, scan_code: 29, repeat_count: 1, key_is_down: false, handled: Handled(false) }
09:43:56.177  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: LeftControl, modifiers: NONE, repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(LeftControl), modifiers: NONE, phys_code: Some(LeftControl), raw_code: 17, scan_code: 29, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
09:43:56.177  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=LeftControl mods=NONE
```
</details>

<details>
<summary>entering a backtick</summary>

```
09:41:49.071  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(LeftShift), modifiers: SHIFT | LEFT_SHIFT, phys_code: Some(LeftShift), raw_code: 16, scan_code: 42, repeat_count: 1, key_is_down: true, handled: Handled(false) }
09:41:49.073  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: LeftShift, modifiers: SHIFT | LEFT_SHIFT, repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(LeftShift), modifiers: SHIFT | LEFT_SHIFT, phys_code: Some(LeftShift), raw_code: 16, scan_code: 42, repeat_count: 1, key_is_down: true, handled: Handled(false) }) }
09:41:49.074  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=LeftShift mods=SHIFT
09:41:49.166  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(RightBracket), modifiers: SHIFT | LEFT_SHIFT, phys_code: Some(RightBracket), raw_code: 221, scan_code: 13, repeat_count: 1, key_is_down: true, handled: Handled(false) }
09:41:49.167  ERROR  window::os::windows::window       > unexpected dead key expansion: [96, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
09:41:49.213  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(RightBracket), modifiers: SHIFT | LEFT_SHIFT, phys_code: Some(RightBracket), raw_code: 221, scan_code: 13, repeat_count: 1, key_is_down: false, handled: Handled(false) }
09:41:49.214  ERROR  window::os::windows::window       > unexpected dead key expansion: [96, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
09:41:49.288  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(LeftShift), modifiers: NONE, phys_code: Some(LeftShift), raw_code: 16, scan_code: 42, repeat_count: 1, key_is_down: false, handled: Handled(false) }
09:41:49.289  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: LeftShift, modifiers: NONE, repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(LeftShift), modifiers: NONE, phys_code: Some(LeftShift), raw_code: 16, scan_code: 42, repeat_count: 1, key_is_down: false, handled: Handled(false) }) }
09:41:49.290  INFO   wezterm_gui::termwindow::keyevent > send to pane UP key=LeftShift mods=NONE
```
</details>

**Comment by wez** [comment]
Would you mind filing a separate issue for the backticks? I suspect that it has something to do with dead keys, especially because of the `unexpected dead key expansion` bit in your logs. In that issue, please also tell me which physical key locations are being pressed: I don't have a german keyboard and I'm not familiar with all of the sequences that are used, and it will make it easier for me to be sure that I'm recreating what you're seeing if you do that!

For the main part of this issue: I'm going to make `allow_win32_input_mode=true` the default; I recently unblocked the last couple of known issues with it in:
* #2009

**Comment by wez** [comment]
Hmm, possibly that is why https://github.com/wez/wezterm/issues/2127 is a problem

**Comment by wez** [comment]
e0616e5eb seems to fix this for me!

**Comment by wez** [comment]
This should be fixed now in `main`.

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out.

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them.

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by wez** [comment]
Good to hear!
I use `fixed-in-nightly` to help me categorize issues that are still open. I manually close them when I think there are no further follow-ups.
